src/orbit_determination/tooshortarc.jl

```diff
@@ -0,0 +1,718 @@
+        # Current Q
+        Q = nms(res)
+        Qs[t] = Q(x0)
+        # Convergence condition
+        t > 1 && abs(Qs[t] - Qs[t-1]) / Qs[t] < Qtol && break
+        # Gradient of objective function
+        g_t = TaylorSeries.gradient(Q)(x0)[5:6]
+        # First momentum
+        m .= μ * m + (1 - μ) * g_t
+        _m_ .= m / (1 - μ^t)
+        # Second momentum
+        n .= ν * n + (1 - ν) * g_t .^ 2
+        _n_ .= n / (1 - ν^t)
+        # Step
+        x1[5:6] = x0[5:6] - η * _m_ ./ (sqrt.(_n_) .+ ϵ)
+        # Update values
+        ρ, v_ρ = bary2topo(A, q(x1))
+        # Projection
+        ρ, v_ρ = boundary_projection(A, ρ, v_ρ)
+        if scale == :log
+            x = log10(ρ)
+        end
+    end
+    # Find point with smallest Q
+    t = argmin(Qs)
+
+    return T(ρs[t]), T(v_ρs[t]), T(Qs[t])
+end
+
+@doc raw"""
+    ρminmontecarlo(radec::Vector{RadecMPC{T}}, A::AdmissibleRegion{T},
+                   params::NEOParameters{T}; N_samples::Int = 25) where {T <: AbstractFloat}
+
+Monte Carlo sampling over the left boundary of `A`.
+"""
+function ρminmontecarlo(radec::Vector{RadecMPC{T}}, A::AdmissibleRegion{T},
+                        params::NEOParameters{T}; N_samples::Int = 25, dynamics::D=newtonian!) where {T <: AbstractFloat, D}
+    # Initial time of integration [julian days]
+    jd0 = datetime2julian(A.date)
+    # Range lower bound
+    ρ = A.ρ_domain[1]
+    # Sample range rate
+    v_ρs = LinRange(A.v_ρ_domain[1], A.v_ρ_domain[2], N_samples)
+    # Allocate memory
+    Qs = fill(Inf, N_samples)
+    # Monte Carlo
+    for i in eachindex(Qs)
+        # Barycentric initial conditions
+        q = topo2bary(A, ρ, v_ρs[i])
+        # Propagation & residuals
+        _, _, res = propres(radec, jd0, q, params; dynamics)
+        iszero(length(res)) && continue
+        # NRMS
+        Qs[i] = nrms(res)
+    end
+    # Find solution with smallest Q
+    t = argmin(Qs)
+
+    return T(ρ), T(v_ρs[t]), T(Qs[t])
+end
+
+@doc raw"""
+    tsals(A::AdmissibleRegion{T}, radec::Vector{RadecMPC{T}}, tracklets::Vector{Tracklet{T}},
+          i::Int, ρ::T, v_ρ::T, params::NEOParameters{T}; maxiter::Int = 5) where {T <: AbstractFloat}
+
+Used within [`tooshortarc`](@ref) to compute an orbit from a point in an
+admissible region via least squares.
+
+!!! warning
+    This function will set the (global) `TaylorSeries` variables to `dx₁ dx₂ dx₃ dx₄ dx₅ dx₆`.
+"""
+function tsals(A::AdmissibleRegion{T}, radec::Vector{RadecMPC{T}}, tracklets::Vector{Tracklet{T}},
+               i::Int, ρ::T, v_ρ::T, params::NEOParameters{T}; maxiter::Int = 5, dynamics::D=newtonian!) where {T <: AbstractFloat, D}
+    # Initial time of integration [julian days]
+    # (corrected for light-time)
+    jd0 = datetime2julian(A.date) - ρ / c_au_per_day
+    # Barycentric initial conditions
+    q0 = topo2bary(A, ρ, v_ρ)
+    # Scaling factors
+    scalings = abs.(q0) ./ 10^5
+    # Jet transport variables
+    dq = scaled_variables("dx", scalings, order = 6)
```

Comment:
Similar to previous comments, I'd suggest to add `order` as a keyword argument to `tsals`, maybe with a different name (`varorder`?) and set `order=varorder` here

Reply:
Solved by https://github.com/PerezHz/NEOs.jl/pull/31/commits/b67b85ec7f3e4b5d0bf443cce1388194d6fbf464.